python/pyspark/sql/_types.py

```diff
@@ -652,7 +652,7 @@ def _python_to_sql_converter(dataType):
 
     if isinstance(dataType, StructType):
         names, types = zip(*[(f.name, f.dataType) for f in dataType.fields])
-        converters = map(_python_to_sql_converter, types)
+        converters = [_python_to_sql_converter(t) for t in types]
```

Comment:
In Python 3, `map` returns a map object instead of a list. So I changed it to `[...]` that is compatible with both 2 & 3.